components/webgpu/lib.rs

```diff
@@ -251,27 +259,26 @@ impl WGPU {
                 },
                 WebGPURequest::CreateBufferMapped(sender, device, id, descriptor) => {
                     let global = &self.global;
-                    let buffer_size = descriptor.size as usize;
-
-                    let (buffer_id, arr_buff_ptr) = gfx_select!(id =>
+                    let (buffer_id, _arr_buff_ptr) = gfx_select!(id =>
                         global.device_create_buffer_mapped(device.0, &descriptor, id));
                     let buffer = WebGPUBuffer(buffer_id);
 
-                    let mut array_buffer = Vec::with_capacity(buffer_size);
-                    unsafe {
-                        array_buffer.set_len(buffer_size);
-                        std::ptr::copy(arr_buff_ptr, array_buffer.as_mut_ptr(), buffer_size);
-                    };
-                    if let Err(e) = sender.send((buffer, array_buffer)) {
+                    if let Err(e) = sender.send(buffer) {
                         warn!(
                             "Failed to send response to WebGPURequest::CreateBufferMapped ({})",
                             e
                         )
                     }
                 },
-                WebGPURequest::UnmapBuffer(buffer) => {
+                WebGPURequest::UnmapBuffer(device_id, buffer, array_buffer) => {
                     let global = &self.global;
-                    gfx_select!(buffer.0 => global.buffer_unmap(buffer.0));
+
+                    gfx_select!(buffer.0 => global.device_set_buffer_sub_data(
```

Comment:
this should only happen if the buffer was mapped for writing

Reply:
We are only call this if the `BufferState` is `MappedForWriting` and the `mapping` slot is an `ArrayBuffer`.